src/measure_theory/card_measurable_space.lean

```diff
@@ -0,0 +1,141 @@
+def generate_measurable_rec (s : set (set α)) : ℵ₁ → set (set α)
+| i := s ∪ {∅} ∪ (λ (t : set α), tᶜ) '' (⋃ j : {j // j <₁ i}, generate_measurable_rec j.1)
+    ∪ (set.range (λ (f : ℕ → (⋃ j : {j // j <₁ i}, generate_measurable_rec j.1)),
+        (⋃ n, (f n).1)))
+using_well_founded {dec_tac := `[exact j.2]}
+
+/-- At each step of the inductive construction, the cardinality bound `≤ 𝔠` is preserved. -/
+lemma cardinal_generate_measurable_rec_le_aux {s : set (set α)} (hs : #s ≤ 𝔠) (i : ℵ₁) :
+  #(generate_measurable_rec s i) ≤ 𝔠 :=
+begin
+  apply well_founded.induction (aleph 1 : cardinal.{u}).ord.out.wo.wf i,
+  assume i IH,
+  have J : #(⋃ (j : {j // j <₁ i}), generate_measurable_rec s j.1) ≤ 𝔠,
+  { apply (mk_Union_le _).trans,
+    have A : # {j // j <₁ i} ≤ aleph 1 := (mk_subtype_le _).trans (le_of_eq (aleph 1).mk_ord_out),
+    have B : (cardinal.sup.{u u} (λ (j : {j // j <₁ i}), #(generate_measurable_rec s j.1)))
+                ≤ 𝔠,
+    { apply cardinal.sup_le.2,
+      rintros ⟨j, hj⟩,
+      exact IH j hj },
+    apply le_trans (mul_le_mul' A B) _,
+    rw mul_eq_max (omega_le_aleph 1) omega_le_continuum,
+    apply max_le aleph_one_le_continuum le_rfl },
+  rw [generate_measurable_rec],
+  apply_rules [(mk_union_le _ _).trans, add_le_of_le omega_le_continuum, mk_image_le.trans,
+    mk_range_le.trans],
+  { rw [mk_singleton],
+    convert (nat_lt_continuum 1).le,
+    simp only [nat.cast_one] },
+  { simp only [mk_pi, subtype.val_eq_coe, prod_const, lift_uzero, mk_denumerable, lift_omega],
+    calc # (↥⋃ (j : {j // j <₁ i}), generate_measurable_rec s ↑j) ^ (ω : cardinal.{u})
+      ≤ 𝔠 ^ (ω : cardinal.{u}) : power_le_power_right J
+    ... = 𝔠 : continuum_power_omega }
+end
+
+lemma cardinal_generate_measurable_rec_le {s : set (set α)} (hs : #s ≤ 𝔠) :
+  #(⋃ i, generate_measurable_rec s i) ≤ 𝔠 :=
+begin
+  have A : (cardinal.sup.{u u} (λ i, #(generate_measurable_rec s i))) ≤ 𝔠 :=
+    cardinal.sup_le.2 (λ i, cardinal_generate_measurable_rec_le_aux hs i),
+  apply (mk_Union_le _).trans,
+  rw [(aleph 1).mk_ord_out],
+  exact le_trans (mul_le_mul' aleph_one_le_continuum A) (by simp)
+end
+
+/-- A set in the sigma-algebra generated by a set of sets `s` is constructed at some step of the
+transfinite induction defining `generate_measurable_rec`. -/
+theorem generate_measurable_subset_rec (s : set (set α)) ⦃t : set α⦄
+  (ht : generate_measurable s t) :
+  t ∈ ⋃ i, generate_measurable_rec s i :=
+begin
+  haveI : nonempty ℵ₁, by simp [← mk_ne_zero_iff, ne_of_gt, (aleph 1).mk_ord_out, aleph_pos 1],
+  inhabit ℵ₁,
+  induction ht with u hu u hu IH f hf IH,
+  { refine mem_Union.2 ⟨default, _⟩,
+    rw generate_measurable_rec,
+    simp only [hu, mem_union_eq, true_or] },
+  { refine mem_Union.2 ⟨default, _⟩,
+    rw generate_measurable_rec,
+    simp only [union_singleton, mem_union_eq, mem_insert_iff, eq_self_iff_true, true_or] },
+  { rcases mem_Union.1 IH with ⟨i, hi⟩,
+    obtain ⟨j, hj⟩ : ∃ j, i <₁ j,
```

Comment:
I believe this should be its own theorem. It's true for any limit ordinal.

Reply:
It already exists (`ordinal.has_succ_of_is_limit`), I've changed the proof to use it.